samza-yarn/src/main/java/org/apache/samza/job/yarn/FsImplConfigManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.job.yarn;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.MapConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * FsImplConfigManager is intented to manage the Samza config for fs.&lt;scheme&gt;impl.
+ * e.g. fs.http.impl
+ *
+ * Usually this config exist in yarn config,
+ * but it is also allowed the clients define this for Samza Job and pick specific implementation for the job.
+ */
+public class FsImplConfigManager {
```

Comment:
@fredji97 : 

0. I did not bring up the inheritance versus composition question in my comment. I'm totally okay with composition (in fact, I'd argue we should favor composition over inheritance). 

1. I 'm slightly opposed to the name - `Manager` in this class when it's just parsing configs and returning them. Usually `Manager`s in Samza do much more than that. For instance, the `ClusterResourceManager` handles life cycle management of all interactions with Yarn, `CheckpointManager` handles checkpoint persistence.